mne/preprocessing/ica.py

```diff
@@ -548,8 +548,7 @@ def _transform_epochs(self, epochs, concatenate):
         if not hasattr(self, 'mixing_matrix_'):
             raise RuntimeError('No fit available. Please fit ICA')
 
-        picks = pick_types(epochs.info, include=self.ch_names, exclude=[],
-                           ref_meg=False)
+        picks = [epochs.ch_names.index(k) for k in self.ch_names]
```

Comment:
it should be more DRY to use the functions rather than the list comprehensions if you think of the functions as being "careful" list comprehensions themselves, so I'd rather use the function if we can. So I guess it might work if you say `meg=False, ref_meg=False, exclude='bads'`?